webrender/src/prim_store.rs

```diff
@@ -1241,36 +1226,26 @@ impl PrimitiveStore {
             cpu_brushes: recycle_vec(self.cpu_brushes),
             cpu_text_runs: recycle_vec(self.cpu_text_runs),
 
-            pictures: recycle_vec(self.pictures),
-            next_picture_id: self.next_picture_id,
-
             chase_id: self.chase_id,
         }
     }
 
-    pub fn add_image_picture(
-        &mut self,
-        composite_mode: Option<PictureCompositeMode>,
-        is_in_3d_context: bool,
-        pipeline_id: PipelineId,
-        reference_frame_index: SpatialNodeIndex,
-        frame_output_pipeline_id: Option<PipelineId>,
-        apply_local_clip_rect: bool,
-    ) -> PictureIndex {
-        let picture = PicturePrimitive::new_image(
-            PictureId(self.next_picture_id),
-            composite_mode,
-            is_in_3d_context,
-            pipeline_id,
-            reference_frame_index,
-            frame_output_pipeline_id,
-            apply_local_clip_rect,
-        );
+    pub fn get_pic(&self, index: PrimitiveIndex) -> &PicturePrimitive {
+        let metadata = &self.cpu_metadata[index.0];
+        let brush = &self.cpu_brushes[metadata.cpu_prim_index.0];
+        match brush.kind {
+            BrushKind::Picture(ref pic) => pic,
+            _ => unreachable!("bug: not a picture!"),
```

Comment:
nit: I think if there is any doubt that this is reachable, then `panic!` is more appropriate :)

Reply:
Done